presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -309,35 +322,39 @@ protected Boolean visitFunctionCall(FunctionCall node, Void context)
                                 "Filtered aggregations not supported with DISTINCT: '%s'",
                                 node);
                     }
+
+                    // ensure that no output fields are referenced from ORDER BY clause
+                    node.getArguments().stream().forEach(expression -> process(expression, true));
```

Comment:
Changed in commit https://github.com/prestodb/presto/pull/7315/commits/912f5d16f91b1249a86540446da8e34080fd27b8